app/models/calculated_value.rb

```diff
@@ -1,3 +1,9 @@
+class String
+    def is_i?
+       /\A[-+]?\d+\z/ === self
```

Comment:
Thanks for catching that.


I need to parse the content of an incoming xml value to see if it contains a number (age).  Could have created a method in CalculatedValue - something like   is_integer(val), but it seems more 'object oriented' to ask the thing itself if it could be an integer. (Maybe there is a sanely-named String method that does this?) I know it seems odd to throw String here.  At some point I'd like to create a separate file that contains convenience methods for classes like String and Integer.


I know this is messy.  Gonna add a TODO to it and keep it for now.


Sheri




________________________________
From: Garrett Martin <notifications@github.com>
Sent: Friday, August 19, 2016 11:52:55 AM
To: SturdyCo/aact2
Cc: Sheri Tibbs; Author
Subject: Re: [SturdyCo/aact2] AACT-166: Add calculated values for min/max age eligibilities. (#81)


In app/models/calculated_value.rb<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_SturdyCo_aact2_pull_81-23discussion-5Fr75506299&d=CwMFaQ&c=imBPVzF25OnBgGmVOlcsiEgHoG1i6YHLR0Sj_gZ4adc&r=UHmIZTL_6Fe-uBOlhrUpgwa0k8zeM9De7MnTpJOTolE&m=-y6nCtg2AYRc2dpbIzgT5lnlQHENQFcPWMoqVF8Vdz4&s=uLgN0CyP__5THooRtkX4pVsgT3x0bai9FMfEtwrREgk&e=>:

> @@ -1,3 +1,9 @@
> +class String
> +    def is_i?
> +       /\A[-+]?\d+\z/ === self


Not sure if this is necessary. String#to_i will return the number inside a string or zero if there is no number inside the string. Not sure that's what you want here, but I wanted to mention it just in case.

-
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_SturdyCo_aact2_pull_81_files_98200acf003e812c83306275f14ddafd3d0be851-23r75506299&d=CwMFaQ&c=imBPVzF25OnBgGmVOlcsiEgHoG1i6YHLR0Sj_gZ4adc&r=UHmIZTL_6Fe-uBOlhrUpgwa0k8zeM9De7MnTpJOTolE&m=-y6nCtg2AYRc2dpbIzgT5lnlQHENQFcPWMoqVF8Vdz4&s=RvZ4KU9CUrfSdrNfsZuvpg8Q5G3SI_-5v-6WWqQyUQQ&e=>, or mute the thread<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ABLfK2MBUZvC1nPMfZ9XmtebG-5FOkFWtRks5qhdFXgaJpZM4JoEac&d=CwMFaQ&c=imBPVzF25OnBgGmVOlcsiEgHoG1i6YHLR0Sj_gZ4adc&r=UHmIZTL_6Fe-uBOlhrUpgwa0k8zeM9De7MnTpJOTolE&m=-y6nCtg2AYRc2dpbIzgT5lnlQHENQFcPWMoqVF8Vdz4&s=x9ntmsfL4EnFOjT0gv8tphp4pB_aR4Xy5vzBuCsO6iQ&e=>.